src/main/kotlin/com/avail/descriptor/representation/AbstractDescriptor.kt

```diff
@@ -0,0 +1,4202 @@
+			}
+			bitFieldsLock.writeLock().lock()
+			return try
+			{
+				// Try again, this time holding the write lock to avoid multiple
+				// threads trying to populate the cache.
+				var bitFields = bitFieldsCache[slot]
+				if (bitFields != null)
+				{
+					return bitFields
+				}
+				val slotAsEnum = slot as Enum<*>
+				val slotClass = slotAsEnum::class.java.declaringClass
+				bitFields = ArrayList()
+				for (field in slotClass.declaredFields)
+				{
+					if (Modifier.isStatic(field.modifiers)
+						&& BitField::class.java.isAssignableFrom(field.type))
+					{
+						try
+						{
+							val bitField = cast<Any, BitField>(field[null])
+							if (bitField.integerSlot === slot)
+							{
+								if (field.getAnnotation(
+										HideFieldInDebugger::class.java)
+									== null)
+								{
+									bitField.enumField = field.getAnnotation(
+										EnumField::class.java)
+									bitField.name = field.name
+									bitFields.add(bitField)
+								}
+							}
+						}
+						catch (e: IllegalAccessException)
+						{
+							throw RuntimeException(e)
+						}
+					}
+				}
+				val sorted =
+					if (bitFields.isEmpty()) emptyList()
+					else bitFields.sorted()
+				bitFieldsCache[slot] = sorted
+				sorted
+			}
+			finally
+			{
+				bitFieldsLock.writeLock().unlock()
+			}
+		}
+
+		/**
+		 * Write a description of an integer field to the [StringBuilder].
+		 *
+		 * @param value
+		 *   The value of the field, a `long`.
+		 * @param numBits
+		 *   The number of bits to show for this field.
+		 * @param enumAnnotation
+		 *   The optional [EnumField] annotation that was found on the field.
+		 * @param builder
+		 *   Where to write the description.
+		 * @throws ReflectiveOperationException
+		 *   If the [EnumField.lookupMethodName] is incorrect.
+		 */
+		@Throws(ReflectiveOperationException::class)
+		private fun describeIntegerField (
+			value: Long,
+			numBits: Int,
+			enumAnnotation: EnumField?,
+			builder: StringBuilder) =
+		with(builder) {
+			if (enumAnnotation != null)
+			{
+				val describingClass: Class<Enum<*>> =
+					cast(enumAnnotation.describedBy)
+				val lookupName = enumAnnotation.lookupMethodName
+				if (lookupName.isEmpty())
+				{
+					// Look it up by ordinal (must be an actual Enum).
+					val allValues: Array<AbstractSlotsEnum> =
+						cast(describingClass.enumConstants)
+					if (0 <= value && value < allValues.size)
+					{
+						append(allValues[value.toInt()].fieldName())
+					}
+					else
+					{
+						append("(enum out of range: ")
+						describeLong(value, numBits, builder)
+						append(")")
+					}
+				}
+				else
+				{
+					// Look it up via the specified static lookup method.  It's
+					// only required to be an IntegerEnumSlotDescriptionEnum in
+					// this case, not necessarily an Enum.
+					val lookupMethod = describingClass.getMethod(
+						lookupName, Int::class.javaPrimitiveType)
+					val lookedUp = lookupMethod.invoke(
```

Comment:
```
when (val lookedUp = lookupMethod.invoke(null, value.toInt())) {
   is IntegerEnumSlotDescriptonEnum? -> ...
   null -> ...
   else -> ...
}
```

Reply:
Took me forever to figure out why copy and paste didn't work. You spelled it `Descripton` instead of `Description`. 🙂

The code of the enclosing `else` block (not shown in your snippet) now reads thus:

```kt
					val lookupMethod = describingClass.getMethod(
						lookupName, Int::class.javaPrimitiveType)
					when (val lookedUp = lookupMethod.invoke(
						null, value.toInt())) {
						is IntegerEnumSlotDescriptionEnum ->
							append(lookedUp.fieldName())
						else ->
							append("null")
					}
```

This is what you wanted, yes?